libs/fabric/src/lib/components/group/directives/group-item.directive.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {
+  AfterContentInit,
+  ContentChildren,
+  Directive,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Output,
+  QueryList,
+} from '@angular/core';
+import { IGroup } from 'office-ui-fabric-react';
+
+import { ItemChangedPayload } from '../../core/declarative/item-changed';
+import { ChangeableItemsHelper, IChangeableItemsContainer } from '../../core/shared/changeable-helper';
+import { ChangeableItemDirective } from '../../core/shared/changeable-item.directive';
+
+@Directive({ selector: 'fab-group-item' })
+export class GroupItemDirective extends ChangeableItemDirective<IGroup>
+  implements AfterContentInit, IChangeableItemsContainer<IGroup>, OnDestroy {
+  @ContentChildren(GroupItemDirective)
+  readonly groupItemsDirectives: QueryList<GroupItemDirective>;
+
+  @Input()
+  name: IGroup['name'];
+  @Input()
+  startIndex: IGroup['startIndex'];
+  @Input()
+  count: IGroup['count'];
+  @Input()
+  children?: IGroup['children'];
+  @Input()
+  level?: IGroup['level'];
+  @Input()
+  isCollapsed?: IGroup['isCollapsed'];
+  @Input()
+  isShowingAll?: IGroup['isShowingAll'];
+  @Input()
+  isDropEnabled?: IGroup['isDropEnabled'];
+  @Input()
+  data?: IGroup['data'];
+  @Input()
+  ariaLabel?: IGroup['ariaLabel'];
+  @Input()
+  hasMoreData?: IGroup['hasMoreData'];
+
+  @Output()
+  get onChildItemChanged(): EventEmitter<ItemChangedPayload<string, IGroup>> {
+    return this.changeableItemsHelper && this.changeableItemsHelper.onChildItemChanged;
+  }
+  @Input()
+  get onItemsChanged(): EventEmitter<QueryList<ChangeableItemDirective<IGroup>>> {
+    return this.changeableItemsHelper && this.changeableItemsHelper.onItemsChanged;
+  }
+
+  private changeableItemsHelper: ChangeableItemsHelper<IGroup>;
```

Comment:
Shouldn't this also use `ChangableItemHelper`? This is why the whole "helper" thing was created (since you essentially needed a way to extend two `class`es).
Or am I missing something?

Reply:
`GroupItemDirective` already extends `ChangeableItemDirective` so only `ChangeableItemsHelper` needed to be used. I think this was the best way to do this since `ChangeableItemsHelper` require the `directiveItems` to be a list of `ChangeableItemDirective`.